python/lsst/cp/pipe/cpFilterScan.py

```diff
@@ -0,0 +1,417 @@
+        dimensions=("instrument", "exposure"),
+        multiple=True,
+        deferLoad=True,
+    )
+    outputData = cT.Output(
+        name="cpFilterScan",
+        doc="Output table to write.",
+        storageClass="ArrowAstropy",
+        dimensions=("instrument", "detector"),
+    )
+
+    def __init__(self, *, config=None):
+        super().__init__(config=config)
+
+        if config and not config.useElectrometer:
+            self.inputs.discard("inputElectrometerHandles")
+
+
+class CpFilterScanTaskConfig(pipeBase.PipelineTaskConfig,
+                             pipelineConnections=CpFilterScanConnections):
+    maskNameList = pexConfig.ListField(
+        dtype=str,
+        doc="Mask list to exclude from statistics calculations.",
+        default=["DETECTED", "BAD", "CR", "NO_DATA", "INTRP"]
+    )
+    useElectrometer = pexConfig.Field(
+        dtype=bool,
+        doc="Use electrometer data?",
+        default=False,
+    )
+    referenceFilter = pexConfig.Field(
+        dtype=str,
+        doc="Filter to use as baseline reference.",
+        default="empty",
+    )
+    combine = pexConfig.ConfigurableField(
+        target=CalibCombineTask,
+        doc="Combine task to use for header merging.",
+    )
+    efdClientInstance = pexConfig.Field(
+        dtype=str,
+        doc="EFD instance to use for monochromator results.",
+        default="usdf_efd",
+    )
+
+
+class CpFilterScanTask(pipeBase.PipelineTask):
+    """Create filter scan from appropriate data.
+
+    This task constructs a filter scan from pairs of flat exposures,
+    one with a filter in the beam and one without a filter.  A
+    monochromator is also included in the beam for both exposures, so
+    that the beam can be limited to a narrow band of wavelengths small
+    than the full bandpass of the filter.
+
+    From these pairs of exposures, we can determine the filter
+    throughput by calculating the flux per second with the filter:
+        F_filter(\lambda0) = median(f_amplifiers) / t_exposure
+    And without:
+        F_reference(\lambda0) = median(f_amplifiers) / t_exposure
+    where the f_amplifiers are the per-amplifier statistics calculated
+    by IsrTask. If the illumination source was perfectly stable, the
+    filter throughput at that wavelength would simply be:
+        throughput_raw(\lambda0) = F_filter / F_reference
+
+    We can correct for any illumination changes by optionally using
+    the electrometer measurements, E, which provide an independent
+    measure of the incident flux for the two exposures, such that:
+        throughput(\lambda0) = throughput_raw * E_reference / E_filter
+
+    Repeating this procedure at multiple monochromator settings builds
+    up a catalog of throughput measurements across the filter
+    bandpass.  Additional differences between the monochromator
+    setting (retrieved here from the EFD) and the actual wavelengths
+    of light that are permitted can exist, so a matching
```

Comment:
"the actual wavelengths of light that are permitted can exist" doesn't make sense.